### PR TITLE
Fix the paths for local stateful storage

### PR DESCRIPTION
**Pull request by srkukarni**

### CODE REVIEW

File: heron/config/src/yaml/conf/local/stateful.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ heron.statefulstorage.classname:            "com.twitter.heron.statefulstorage.l
 
 heron.statefulstorage.config:
   heron.statefulstorage.classpath:           ""
-  heron.statefulstorage.localfs.root.path:   "./checkpoints"
+  heron.statefulstorage.localfs.root.path:   ~/.herondata/checkpoints
```
Comment (kramasamy) [inline heron/config/src/yaml/conf/local/stateful.yaml]: you should use ${HOME}/.herondata/checkpoints

File: heron/config/src/yaml/conf/localzk/stateful.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ heron.statefulstorage.classname:            "com.twitter.heron.statefulstorage.l
 
 heron.statefulstorage.config:
   heron.statefulstorage.classpath:           ""
-  heron.statefulstorage.localfs.root.path:   "./checkpoints"
+  heron.statefulstorage.localfs.root.path:   ~/.herondata/checkpoints
```
Comment (kramasamy) [inline heron/config/src/yaml/conf/localzk/stateful.yaml]: you should use ${HOME}/.herondata/checkpoints

File: heron/config/src/yaml/conf/local/stateful.yaml (hunk 1 above)
Comment (srkukarni) [inline heron/config/src/yaml/conf/local/stateful.yaml]: It's much easier to expand ~ than using ${HOME}.
BTW the other uses of ${HOME} used in files like statemgr.yaml is really not used by the code. Its being overriden manually. See https://github.com/twitter/heron/blob/master/heron/statemgrs/src/java/com/twitter/heron/statemgr/localfs/LocalFileSystemStateManager.java#L245
I will fix this as well in another pr

**Comment by srkukarni** [comment]
Related to #2146

**Comment by srkukarni** [comment]
https://github.com/twitter/heron/issues/2146 will track moving away from CheckpointManagerConfig to the generic Config class.

**Pull request merged**